src/components/EditTask.js

```diff
@@ -30,22 +29,38 @@ class EditTask extends PureComponent {
     event.preventDefault()
 
     const { edit, history } = this.props
+    const { id, comment, comments } = this.state.task
+
+    const comments_ = !comment
+      ? [...comments]
+      : comments.length > 0
+      ? [...comments, comment]
+      : [comment]
 
     const task = {
       ...this.state.task,
-      done: false,
+      comments: comments_,
     }
 
-    const { id } = this.state.task
+    delete task.comment
 
     edit(id, task)
     history.push('/list/')
   }
 
   handleClearForm = () => {
-    this.setState({
-      task: { name: '', description: '', comment: '', priority: '' },
-    })
+    const cleanForm = {
+      name: '',
+      description: '',
+      comment: '',
+      priority: '',
+    }
+
+    const task = {
+      ...this.state.task,
+      ...cleanForm,
+    }
+    this.setState({ task })
   }
 
   componentDidMount() {
```

Comment:
Think about https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops instead of CompoenntDidMount()